### DISCUSSION TITLE
make chansrv_fuse multisession compatible when running as the same user

### DISCUSSION THREAD

**Discussion by akarl10** [neutrinolabs/xrdp#2525] [General]
Hello,

I am currently experimenting with xrdp running multiple sessions as the same user.
One thing that does not really play well with this kind of setup is that currently there is no way to create separate fuse root directories

To fix this I made a some experiments here https://github.com/akarl10/xrdp/commits/chansrv-display-variable

Doing this I saw that the variable replacement is done only if the path is absolute. Is there a specific reason why other than previously it did non make a lot of sense to use uid and username in the home path?

**Comment by matt335672** [comment]
Original PR is neutrinolabs/xrdp#1703.

I can't remember any specific reason why we didn't do the subs for the relative option. If we're going to add other options (%d for display maybe?), it would make sense to change this I think.

**Reply by akarl10** [reply]
What could be the decision why this is used only in absolute paths is that if the HOME path contains % this must not be escaped.
This would necessitate to escape the relative path and prepend HOME and considering there was not really a usecase for this the escaping was not done at all
